### PR TITLE
check in the files for packer build:

### PR DESCRIPTION
**Pull request by PaynePei**
including:
1.two json files for box and ova build 
2.one vagrantfile for box deploy 
3.one configure file for os install 
4.some shell scripts for software install and upgrade

### CODE REVIEW

**Comment by PaynePei** [comment]
@xiar @MarkMa001 @fub2 @turtle-fly @JuneZhou19 @XiaowenJiang please help review

File: packer/scripts/infrasim-compute.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+export LC_ALL=C
+
+# instal dependency for infrasim-compute
+apt-get -y install socat ipmitool qemu openipmi python-pip libssl-dev libssh-dev libpython-dev libffi-dev sgabios
```
Comment (turtle-fly) [inline packer/scripts/infrasim-compute.sh]: libssl-dev is installed in base.sh

**Comment by turtle-fly** [comment]
我们可以省下时间来嗑瓜子了，good job

File: packer/Vagrantfile [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+# backwards compatibility). Please don't change it unless you know what
+# you're doing.
+Vagrant.configure(2) do |config|
+  # The most common configuration options are documented and commented below.
+  # For a complete reference, please see the online documentation at
+  # https://docs.vagrantup.com.
+
+  # Every Vagrant development environment requires a box. You can search for
+  # boxes at https://atlas.hashicorp.com/search.
+  config.vm.box = "infrasim"
```
Comment (fub2) [inline packer/Vagrantfile]: how about infrasim-compute?

File: packer/Vagrantfile [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  config.vm.provider "virtualbox" do |vb|
+    vb.memory = 4096
+    vb.cpus = 4
+    vb.customize ["modifyvm", :id, "--ioapic", "on", "--cpus", "4"]
+  end
+
+  config.vm.base_mac = "080027627FDA"
```
Comment (fub2) [inline packer/Vagrantfile]: Would suggest at least giving 2 NICs for infrasim running on virtualbox: first one for purpose of NAT connection occupied by vagrant/virtualbox, second one for control/data network of vritual node.

File: packer/Vagrantfile (hunk 2 above)
Comment (PaynePei) [inline packer/Vagrantfile]: sure, user should keep align with the command "vagrant box add --name ******"

File: packer/Vagrantfile [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+  config.vm.network "forwarded_port", guest: 5901, host: 15901
```
Comment (fub2) [inline packer/Vagrantfile]: Can you added comments on purpose of forwarding these ports?

**Comment by PaynePei** [comment]
@fub2 one more network is added for ova and virtualbox separately

File: packer/Vagrantfile (hunk 3 above)
Comment (PaynePei) [inline packer/Vagrantfile]: fixed

File: packer/scripts/infrasim-compute.sh (hunk 1 above)
Comment (PaynePei) [inline packer/scripts/infrasim-compute.sh]: keep align with the docs

**Comment by turtle-fly** [comment]
OK to me

**Comment by fub2** [comment]
+1

**Pull request merged**